lib/auth/auth_with_roles.go

```diff
@@ -5627,7 +5702,25 @@ func (a *ServerWithRoles) DeleteAllSAMLIdPServiceProviders(ctx context.Context)
 		return trace.Wrap(err)
```

Comment:
Technically, if someone without adequate permissions tries to delete all providers, we won't audit the failed attempt. Seems like something worth addressing.